### ISSUE TITLE
show size of download in the Contribution Manager

### ISSUE THREAD

**Issue by benfry** [processing/processing#3926]

**Comment by iamutkarshtiwari** [comment]
I am currently working on this enhancement.

**Comment by iamutkarshtiwari** [comment]
Adding this functionality introduces a bit ***lag*** in Contribution Manager b/c everytime we click a contributions, the will need fetching and ***calculating the size*** of the download file which takes a little time. Is this lag be acceptable?

**Comment by benfry** [comment]
That should be handled on the server side instead, when the contrib list is built. That's a lot of connections and a lot of lag. @prisonerjohn might have thoughts on that.

**Comment by iamutkarshtiwari** [comment]
@benfry I somehow figured out how to handle that so that it may not end up introducing a lag in the CM. The above patch handles that problem and also integrates the "Download FileSize" feature into CM.

**Comment by iamutkarshtiwari** [comment]
@benfry After the above patch, the ***show dowload size*** feature will show up in the CM as shown in the screenshot below.
![screen](https://cloud.githubusercontent.com/assets/6258810/10713928/e1f02120-7af8-11e5-82d7-dd6391caba77.png)

**Comment by DarkPrince304** [comment]
Hello,

You mentioned that something had to be done on the server-side of the contribs file. Did you mean to make a size parameter for each entry and then retrieve it like everything else is retrieved on client-side (PDE Manager) ?

Like this:

"
library
minRevision=0
name=Simple Multi-Touch (SMT)
maxRevision=228
sentence=Multi-touch prototyping and development made simple.
url=http://vialab.science.uoit.ca/SMT/
prettyVersion=4.2
paragraph=SMT provides back-end device support for Tuio, Windows Touch* ( requires [VC2012 Redistributable](http://www.microsoft.com/en-au/download/details.aspx?id=30679) ), SmartSDK*, Leap Motion*, and Mouse Emulation. *Windows-Only
version=23
authors=[Erik Paluka](http://www.erikpaluka.com/), [Kalev Kalda Sikes](http://vialab.science.uoit.ca/portfolio/kalev-kalda-sikes), [Zachary Cook](http://vialab.science.uoit.ca/portfolio/zachary-cook), [Dr. Mark Hancock](http://markhancock.ca/), and [Dr. Christopher Collins](http://vialab.science.uoit.ca/portfolio/christopher-m-collins)
download=http://vialab.science.uoit.ca/smt/dl/SMT.zip
type=library
id=098
size=1024  <----------------
categories=I/O
"

Thanks!

**Comment by benfry** [comment]
Yes, or better, that it'd be added to the file automatically when @prisonerjohn is building the file. But that's be his call about his process for how the file is built.

**Comment by prisonerjohn** [comment]
Agreed that this should be handled by the server script. I started looking into this a few weeks ago but haven't had the time to finish it. 

I'll assign to myself for now, but the script is here if anyone wants to take a stab at it: https://github.com/processing/processing-docs/blob/master/contrib_generate/build_listing.py
